test/logic/stake_logic_test.rb

```diff
@@ -0,0 +1,93 @@
+require 'test_helper'
+
+# StakeLogicTest
+class StakeLogicTest < ActiveSupport::TestCase
+  include StakeLogic
+
+  def setup
+    @testnet_url = 'https://api.testnet.solana.com'
+    @mainnet_url = 'https://api.mainnet-beta.solana.com'
+
+    @batch = create(:batch, network: 'testnet', gathered_at: DateTime.now, scored_at: DateTime.now)
+
+    # Create our initial payload with the input values
+    @initial_payload = {
+      # config_urls: Rails.application.credentials.solana[:testnet_urls],
+      config_urls: [
+        @testnet_url
+      ],
+      network: 'testnet'
+    }
+  end
+
+  test 'get_last_batch' do
+    p = Pipeline.new(200, @initial_payload)
+                .then(&get_last_batch)
+    assert_not_nil p[:payload][:batch]
+    assert p[:payload][:batch].uuid.include?('-')
+  end
+
+  test 'move_current_stakes_to_history' do
+    create(:stake_account, batch_uuid: 'old-batch')
+    json_data = File.read("#{Rails.root}/test/json/stake_accounts.json")
```

Comment:
```suggestion
    json_data = file_fixture('json/stake_accounts.json')
```
if you move file to `test/fixtures/files/json`. Same for rest tests. 
Also, since it reapeats in tests it could be instance variable.